api/v1beta1/activemqartemis_types.go

```diff
@@ -575,16 +575,42 @@ type ActiveMQArtemisStatus struct {
 	//see 3scale https://github.com/3scale/3scale-operator/blob/8abbabd926616b98db0e7e736e68e5ceba90ed9d/apis/apps/v1alpha1/apimanager_types.go#L87
 
 	// Current state of external referenced resources
-	//+optional
 	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="External Configurations Status"
 	ExternalConfigs []ExternalConfigStatus `json:"externalConfigs,omitempty"`
+
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="Version Status"
+	Version VersionStatus `json:"version,omitempty"`
+}
+
+type VersionStatus struct {
+
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="BrokerVersion",xDescriptors="urn:alm:descriptor:text"
+	BrokerVersion string `json:"brokerVersion,omitempty"`
+
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="Image URI",xDescriptors="urn:alm:descriptor:org.w3:link"
+	Image string `json:"image,omitempty"`
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="InitImage URI",xDescriptors="urn:alm:descriptor:org.w3:link"
+	InitImage string `json:"initImage,omitempty"`
+
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="Upgrade Status"
+	Upgrade UpgradeStatus `json:"upgrade,omitempty"`
```

Comment:
Should Upgrade be a property of ActiveMQArtemisStatus?

Reply:
maybe, it has Status as the postfix. I think that would be better, it is not just a function of Version. thanks!